Makefile

```diff
@@ -15,21 +15,32 @@ check-dependencies:
 	@#(call check-dependency,cut)
 	@#(call check-dependency,sed)
 
+CP_VERSION ?= 5.3.1
+KAFKA_CONNECT_DATAGEN_VERSION ?= 0.1.5
+AGGREGATE_VERSION = $(KAFKA_CONNECT_DATAGEN_VERSION)-$(CP_VERSION)
+
+KAFKA_CONNECT_DATAGEN_LOCAL_VERSION = $(shell make local-package-version)
+AGGREGATE_LOCAL_VERSION = $(KAFKA_CONNECT_DATAGEN_LOCAL_VERSION)-$(CP_VERSION)
+
 help:
 	@$(foreach m,$(MAKEFILE_LIST),grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(m) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-$(HELP_TAB_WIDTH)s\033[0m %s\n", $$1, $$2}';)
 
-version-connector: check-dependencies
-	@grep COPY Dockerfile-dockerhub | cut -d':' -f2 | sed 's/.*datagen-\(.*\)\.zip/\1/'
+local-package-version: ## Retrieves the jar version from the maven project definition
+	@mvn help:evaluate -Dexpression=project.version -q -DforceStdout
+
+package: check-dependencies ## Creates the assembly jar
+	@mvn clean package
 
-version-connect: check-dependencies
-	@grep FROM Dockerfile-dockerhub | cut -d':' -f2
+build-local: ## Build the Docker image using the locally mvn built kafka-connect-datagen package
+	@docker build -t kafka-connect-datagen:$(AGGREGATE_LOCAL_VERSION) --build-arg KAFKA_CONNECT_DATAGEN_VERSION=$(KAFKA_CONNECT_DATAGEN_LOCAL_VERSION) --build-arg CP_VERSION=$(CP_VERSION) -f Dockerfile .
 
-version: ## Show the current package version of the project
-	@echo $(shell make version-connector)-$(shell make version-connect)
+build-cp-kafka-connect-confluenthub: ## Build the cp-kafka-connect image pulling datagen from Confluent Hub
+	@docker build -t cnfldemos/kafka-connect-datagen:$(AGGREGATE_VERSION) --build-arg KAFKA_CONNECT_DATAGEN_VERSION=$(KAFKA_CONNECT_DATAGEN_VERSION) --build-arg CP_VERSION=$(CP_VERSION) -f Dockerfile-confluenthub .
 
-package: check-dependencies ## Creates any package artifacts (Docker, Assembly JAR, etc..)
-	@mvn clean package
+build-cp-server-connect-operator-confluenthub: ## Build the cp-server-connect-operator image pulling datagen from Confluent Hub
+	@docker build -t cnfldemos/cp-server-connect-operator-with-datagen:$(AGGREGATE_VERSION) --build-arg KAFKA_CONNECT_DATAGEN_VERSION=$(KAFKA_CONNECT_DATAGEN_VERSION) --build-arg CP_VERSION=$(CP_VERSION) -f Dockerfile-operator .
+
+publish: build-cp-kafka-connect-confluenthub build-cp-server-connect-operator-confluenthub ## Builds and publishes all images to Dockerhub
```

Comment:
Is the idea to publish cnfldemos/kafka-connect-datagen and cnfldemos/cp-server-connect-operator-with-datagen simultaneously all the time, and they cannot be done independently?  I don't think we can or should tie these together -- we should be able to do them independently.  Operator image releases may lag Docker image releases.

Reply:
Agree, I'll remove the aggregate publish and add a docker push to each of the individual build targets